src/paperless/views.py

```diff
@@ -168,3 +172,57 @@ class ApplicationConfigurationViewSet(ModelViewSet):
 
     serializer_class = ApplicationConfigurationSerializer
     permission_classes = (IsAuthenticated, DjangoObjectPermissions)
+
+
+class DisconnectSocialAccountView(GenericAPIView):
+    """
+    Disconnects a social account provider from the user account
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, *args, **kwargs):
+        user = self.request.user
+
+        try:
+            account = user.socialaccount_set.get(pk=request.data["id"])
+            account_id = account.id
+            account.delete()
+            return Response(account_id)
+        except SocialAccount.DoesNotExist:
+            return HttpResponseBadRequest("Social account not found")
+
+
+class SocialAccountProvidersView(APIView):
+    """
+    List of social account providers
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request, *args, **kwargs):
+        adapter = get_adapter()
+        providers = adapter.list_providers(request)
+        resp = [
+            {"name": p.name, "login_url": p.get_login_url(request, process="connect")}
+            for p in providers
+            if p.id != "openid"
+        ]
+
+        if (
+            openid_provider := next(filter(lambda p: p.id == "openid", providers), None)
+            is not None
+        ):
+            resp += [
+                {
+                    "name": b.name,
+                    "login_url": openid_provider.get_login_url(
+                        request,
+                        process="connect",
+                        openid=b.openid_url,
+                    ),
+                }
+                for b in openid_provider.get_brands()
+            ]
```

Comment:
Whats the purpose of this special case? And please add coverage for it, this is the only remaining uncovered line

Reply:
It's the equivalent to what allauth is doing in their template: https://github.com/pennersr/django-allauth/blob/main/allauth/templates/socialaccount/snippets/provider_list.html
The `openid` provider is a bit special and configured slightly different compared to the others: https://docs.allauth.org/en/latest/socialaccount/providers/openid.html
I'll see to add a test for those lines as well.

Reply:
I couldn't resist. I also dont think that code was working (hence, tests 👍)

`PAPERLESS_APPS="allauth.socialaccount.providers.openid"`
<img width="653" alt="Screenshot 2024-02-05 at 1 04 37 AM" src="https://github.com/paperless-ngx/paperless-ngx/assets/4887959/bf018805-202c-4de4-b198-bb28075fe01f">

Reply:
Ah, I think a pair of parenthesis was missing. :see_no_evil:  Nice, thanks for adding the test